scroll-search.lic

```diff
@@ -0,0 +1,520 @@
+	
+	echo ("*** SEARCH COMPLETED ***")
+  end
+
+  def display_scrolls()
+    current_scrolls = UserVars.owned_scrolls
+
+    if current_scrolls.length <= 0 || current_scrolls.nil?
+      echo ("*** NO SCROLLS TO DISPLAY -- SEARCH FIRST! ***")
+      return
+    end
+
+	
+    echo ("|============================== Scroll List =============================|")
+    echo (Kernel::sprintf("|%s%s%s%s%s%s%s   %s", format_bar(3, "#"), format_bar(3, "|"), format_bar(14, "Scroll Name"), format_bar(7, "|"),
+																	  format_bar(current_scrolls[0].get_container.length - "Container".length, "Container"), format_bar("Container|".length, "|"), 
+																	  format_bar(5 + "Type".length, "Type"), format_bar("Type".length, "|")))
+    echo ("|========================================================================|")
+
+    # Iterate over scrolls to display
+    index = 0
+
+    current_scrolls.each do |container, scroll_array|
+	  
+	  scroll_array.each do |scroll_info|
+		the_scroll = Kernel::sprintf("|%s%s%s%s%s%s%s%s", format_bar(3, scroll_info.get_index), format_bar(3, "|"), format_bar(14, scroll_info.get_spell), 
+																			format_bar(7, "|"), format_bar(-7, scroll_info.get_container), format_bar(-5, "|"), 
+																			format_bar(7, scroll_info.get_type), format_bar(5, "|"))
+
+		echo(the_scroll)
+		index += 1
+	  end
+    end
+
+    echo ("|                                                                        |")
+    echo ("|           (;send <get #> or ;send <transfer # to <container>>)         |")
+    echo ("|========================================================================|")
+
+    line = nil
+    line = get until line.strip =~ /^get\s+(\d+)$|^tran.+\s+(\d+)\s+to\s+(\w+)$/i
+
+    case line
+    when /^get\s+(\d+)$/i
+      scroll_index = Regexp.last_match(1)
+      echo("Scroll index is #{scroll_index}.") if $debug_mode
+
+      grab_scroll(scroll_index)
+    when /^tran.+\s+(\d+)\s+to\s+(\w+)$/i
+      scroll_index = Regexp.last_match(1)
+      transfer_container = Regexp.last_match(2)
+
+      echo("Index is #{scroll_index}. Transfer is: #{transfer_container}") if $debug_mode
+
+      transfer_scroll(scroll_index, transfer_container)
+    end
+  end
+
+  # Private methods
+  private
+
+  def get_scrolls(scroll)
+    scroll_noun = scroll #get_noun(scroll)
+    parameter = "C #{scroll_noun}"
+    scrolls = rummage(parameter, $container)
+    return scrolls
+  end
+
+  def look_scrolls(all_scrolls)
+    # Reset scroll count
+	index = 0
+
+    all_scrolls.each do |scroll|
+      ordinal = $ORDINALS[index]
+
+      case scroll_look = bput("look #{ordinal} #{scroll} in my #{$container}", 'It is labeled "\w*\s*\w*."', '.* of the \w*\s*\w* spell.', 'three-dimensional shapes cover much of the scroll', 'You see nothing unusual.', 'I could not find what you were referring to')
+      when 'You see nothing unusual.', 'What were you referring to', 'I could not find what you were referring to' then break
```

Comment:
It might be easier to make a when for `/It is labeled "(\w*\s*\w*)."/i` and remove this when, letting all of these cases just be ignored.